dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java

```diff
@@ -220,6 +240,51 @@ private List<FileInfo> getFilePreviewContent(Context context, Bitstream bitstrea
         return fileInfos;
     }
 
+    /**
+     * Create sub map for preview content and file info.
+     * @param sourceMap  parent sub map
+     * @param creator    creator function
+     * @return    created sub map
+     */
+    private <T, U> Hashtable<String, T> createSubMap(Map<String, U> sourceMap, Function<U, T> creator) {
+        Hashtable<String, T> sub = new Hashtable<>();
+        if (sourceMap != null) {
+            for (Map.Entry<String, U> entry : sourceMap.entrySet()) {
+                sub.put(entry.getKey(), creator.apply(entry.getValue()));
+            }
+        }
+        return sub;
+    }
+
+    /**
+     * Create file info from preview content.
+     * @param pc  preview content
+     * @return    created file info
+     */
+    private FileInfo createFileInfo(PreviewContent pc) {
+        Hashtable<String, FileInfo> sub = createSubMap(pc.sub, this::createFileInfo);
+        return new FileInfo(pc.name, pc.content, pc.size, pc.isDirectory, sub);
+    }
+
+    /**
+     * Create preview content from file info for bitstream.
+     * @param context   DSpace context object
```

Comment:
In some classes you have an empty line between the comment header and the `@param` I suggest you to be have it consistent.

Reply:
They use empty lines, so I used them in most of the created classes. However, the class MetadataBitstreamRestRepository was created by you, and you didn't use empty lines. For consistency, I think I should not use empty lines in this class.